gufe/components/proteincomponent.py

```diff
@@ -210,36 +265,38 @@ def _from_openmmPDBFile(cls, openmm_PDBFile: Union[PDBFile, PDBxFile], name: str
         # Add Additionals
         # Formal Charge
         netcharge = 0
-        for a in rd_mol.GetAtoms():
-            atomic_num = a.GetAtomicNum()
-            atom_name = a.GetMonomerInfo().GetName()
-
-            connectivity = sum(_BONDORDER_TO_ORDER[bond.GetBondType()] for bond in a.GetBonds())
-            default_valence = periodicTable.GetDefaultValence(atomic_num)
-
-            if connectivity == 0:  # ions:
-                # strip catches cases like 'CL1' as name
-                if atom_name.strip(string.digits).upper() in positive_ions:
-                    fc = default_valence  # e.g. Sodium ions
-                elif atom_name.strip(string.digits).upper() in negative_ions:
-                    fc = -default_valence  # e.g. Chlorine ions
-                else:  # -no-cov-
-                    resn = a.GetMonomerInfo().GetResidueName()
-                    resind = int(a.GetMonomerInfo().GetResidueNumber())
-                    raise ValueError(
-                        "I don't know this Ion or something really went "
-                        f"wrong! \t{atom_name}\t{resn}\t-{resind}\t"
-                        f"connectivity{connectivity}"
-                    )
-            elif default_valence > connectivity:
-                fc = -(default_valence - connectivity)  # negative charge
-            elif default_valence < connectivity:
-                fc = +(connectivity - default_valence)  # positive charge
-            else:
-                fc = 0  # neutral
+        for omm_atom, rd_atom in zip(mol_topology.atoms(), rd_mol.GetAtoms()):
+            # Transfer formal charge from OpenMM atom to RDKit atom if it exists
 
-            a.SetFormalCharge(fc)
-            a.UpdatePropertyCache(strict=True)
+            if hasattr(omm_atom, "formalCharge") and omm_atom.formalCharge is not None:
+                fc = omm_atom.formalCharge
+            else:
+                atom_name = rd_atom.GetMonomerInfo().GetName().strip()
+                atomic_num = rd_atom.GetAtomicNum()
+
+                connectivity = sum(_BONDORDER_TO_ORDER[bond.GetBondType()] for bond in rd_atom.GetBonds())
+                default_valence = periodicTable.GetDefaultValence(atomic_num)
+
+                if connectivity == 0:  # ions
+                    ion_key = atom_name.strip().upper()
```

Comment:
We need to make sure we catch cases like CL1 and strip digits from the atom name similarly to the old method.